lib/galaxy/tools/__init__.py

```diff
@@ -622,6 +622,21 @@ def parse(self, tool_source, guid=None):
 
         self.parse_command(tool_source)
         self.environment_variables = self.parse_environment_variables(tool_source)
+        self.tmp_directories = tool_source.parse_tmp_directories()
+
+        home_target = tool_source.parse_home_target()
+        tmp_target = tool_source.parse_tmp_target()
+        # If a tool explicitly sets one of these variables just respect that and turn off
+        # explicit processing by Galaxy.
+        for environment_variable in self.environment_variables:
+            if environment_variable.get("name") == "HOME":
+                home_target = None
+            for tmp_directory in self.tmp_directories:
+                if environment_variable.get("name") == tmp_directory:
+                    tmp_target = None
```

Comment:
If `tmp_target` is None you don't need this for loop anymore.

Reply:
+1